src/xpk/commands/job.py

```diff
@@ -14,12 +14,87 @@
 limitations under the License.
 """
 
+from ..core.commands import run_command_for_value, run_command_with_updates
 from ..utils.console import xpk_exit, xpk_print
-from ..core.core import add_zone_and_project
 from ..core.app_profile import APP_PROFILE_TEMPLATE_DEFAULT_NAME
-from ..core.commands import (
-    run_command_with_updates,
-)
+from ..core.core import add_zone_and_project
+from ruamel.yaml import YAML
+import re
+import sys
+
+
+def job_info(args):
+  """Run commands obtaining information about a job given by name.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  job_name = args.name
+
+  desc_command = f'kubectl-kjob describe slurm {job_name}'
+  desc_code, desc_text = run_command_for_value(
+      desc_command, 'Getting job data', args
+  )
+  if desc_code != 0:
+    xpk_print(f'Data info request returned ERROR {desc_code}')
+    xpk_exit(desc_code)
+
+  job_command = (
+      'kubectl-kjob list slurm -o yaml --field-selector'
+      f' metadata.name=={job_name}'
+  )
+  job_code, job_text = run_command_for_value(
+      job_command, 'Getting job info', args
+  )
+  if job_code != 0:
+    xpk_print(f'Job info request returned ERROR {job_code}')
+    xpk_exit(job_code)
+
+  pods_command = f'kubectl get pods -l=job-name={job_name} --no-headers'
+  pods_code, pods_text = run_command_for_value(
+      pods_command, 'Getting pods list', args
+  )
+  if pods_code != 0:
+    xpk_print(f'Pods list request returned ERROR {pods_code}')
+    xpk_exit(pods_code)
+
+  yaml = YAML(typ='safe')
+  job_yaml = yaml.load(job_text)['items'][0]
+
+  output = {
+      'Job name': job_name,
+      'Profile': get_profile(job_yaml),
+      'Labels': job_yaml['metadata']['labels'],
+      'Mounts': job_yaml['spec']['template']['spec']['containers'][0][
+          'volumeMounts'
+      ],
+      'Environment variables': get_ev_vars(desc_text),
+      'Pods': get_pods(pods_text),
+  }
+
+  yaml.default_flow_style = False
+  yaml.sort_keys = False
+  yaml.dump(output, sys.stdout)
+
+
+def get_profile(job_yaml: dict) -> str:
+  env_vars = job_yaml['spec']['template']['spec']['containers'][0]['env']
+  profile = next((x['value'] for x in env_vars if x['name'] == 'PROFILE'), '')
+  return profile
+
+
+def get_ev_vars(job_desc_text: str) -> list[tuple[str, str]]:
+  regex = r'(SLURM_[A-Z_]*=.*)'
```

Comment:
It will not work well. We can't see env variables with describe, because it is generating on runtime.